python/triton/runtime/cache.py

```diff
@@ -258,7 +258,18 @@ def put_group(self, filename: str, group: Dict[str, str]):
 
 def _base64(key):
     # Assume key is a hex string.
-    return base64.urlsafe_b64encode(bytes.fromhex(key)).decode("utf-8").rstrip("=")
+    encoded = base64.urlsafe_b64encode(bytes.fromhex(key)).decode("utf-8").rstrip("=")
```

Comment:
`urlsafe` will generate those `-` instead of `+`, one simpler way is to just use a different replacement character
I think this should work:
```
return base64.b64encode(bytes.fromhex(key), altchars=b'._').decode('utf-8').rstrip('=')
```

Reply:
Thanks for the review !

It's a good approach but I think using a dot in this context might cause confusion, as it's usually interpreted as separating a filename from its extension.

Reply:
you're right sorry I meant to use a `+` instead of a `.`

Reply:
No worries at all and thanks again for your review !

In this way, we lose the benefits of using the urlsafe.

If you think that approach is better, I can make the changes.

Let me know !

Reply:
yeah I don't think it needs to url safe?

Reply:
otherwise we could go back to base32, the names will be a bit longer but I think it is fine too

Reply:
> yeah I don't think it needs to url safe?

If the cache files aren’t intended for use in URLs, making them urlsafe is not necessary.
However, I’d prefer to keep them URL-safe to retain the option for future flexibility.

> otherwise we could go back to base32, the names will be a bit longer but I think it is fine too

Using Base32 does lose some of the compactness we gain with Base64. If I’m not mistaken, Base32 would give us a 52-character string, compared to 64 characters with the SHA-256 hex format and 44 characters with Base64.

Let me know and thanks again!

Reply:
the special characters are bit annoying to me so I wouldn't mind going to Base32

Reply:
oh, okay!
I thought you were ok with it since you approved the #4553

Reply:
yeah I didn't realize that meant there would be those special characters